Update WPT mediacapture-streams tests up to b7e7973

youennf opened the pull request:
#### 517a80a70630302d898acece2b0f6de72127558c
<pre>
Update WPT mediacapture-streams tests up to b7e7973
<a href="https://bugs.webkit.org/show_bug.cgi?id=256782">https://bugs.webkit.org/show_bug.cgi?id=256782</a>
rdar://problem/109349894

Reviewed by Eric Carlson.

Update testRunner API to allow setting either microphone or camera permission independently.
Make use of this API in testdriver-vendor.js.
We still keep testRunner.setUserMediaPermission as a shortcut for both camera and microphone.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/mediacapture-streams: Resynced.
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(async switch):
(window.test_driver_internal.set_permission):
* LayoutTests/platform/glib/TestExpectations:
* Source/WebKit/UIProcess/API/C/WKUserMediaPermissionRequest.cpp:
(WKUserMediaPermissionRequestRequiresCameraCapture):
(WKUserMediaPermissionRequestRequiresMicrophoneCapture):
* Source/WebKit/UIProcess/API/C/WKUserMediaPermissionRequest.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::setCameraPermission):
(WTR::InjectedBundle::setMicrophonePermission):
(WTR::InjectedBundle::setUserMediaPermission): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setCameraPermission):
(WTR::TestRunner::setMicrophonePermission):
(WTR::TestRunner::setUserMediaPermission):
(WTR::TestRunner::resetUserMediaPermission):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::setCameraPermission):
(WTR::TestController::setMicrophonePermission):
(WTR::TestController::decidePolicyForUserMediaPermissionRequestIfPossible):
(WTR::TestController::setUserMediaPermission): Deleted.
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/264149@main">https://commits.webkit.org/264149@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/23923b7df864fd4697fa57255c9d939e1e323b67

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6865 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7094 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7278 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8463 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7112 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6869 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8334 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7034 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10010 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6986 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7645 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6313 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8562 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/5038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6228 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14014 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6313 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9097 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6796 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5562 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6173 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/6142 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1626 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10357 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6551 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -2770,7 +2780,12 @@ void TestController::decidePolicyForUserMediaPermissionRequestIfPossible()
         auto& settings = settingsForOrigin(originHash);
         settings.incrementRequestCount();
 
-        if (!m_isUserMediaPermissionAllowed && !settings.persistentPermission()) {
```
Is persistent permission no longer supported?

youennf replied:
persistentPermission is limited to enumerateDevices showing deviceIds in that case.
We should deprecate that code path since the spec mandates exposing deviceId once getUserMedia is successfully called.

youennf committed:
Update WPT mediacapture-streams tests up to b7e7973
https://bugs.webkit.org/show_bug.cgi?id=256782
rdar://problem/109349894

Reviewed by Eric Carlson.

Update testRunner API to allow setting either microphone or camera permission independently.
Make use of this API in testdriver-vendor.js.
We still keep testRunner.setUserMediaPermission as a shortcut for both camera and microphone.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/mediacapture-streams: Resynced.
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(async switch):
(window.test_driver_internal.set_permission):
* LayoutTests/platform/glib/TestExpectations:
* Source/WebKit/UIProcess/API/C/WKUserMediaPermissionRequest.cpp:
(WKUserMediaPermissionRequestRequiresCameraCapture):
(WKUserMediaPermissionRequestRequiresMicrophoneCapture):
* Source/WebKit/UIProcess/API/C/WKUserMediaPermissionRequest.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::setCameraPermission):
(WTR::InjectedBundle::setMicrophonePermission):
(WTR::InjectedBundle::setUserMediaPermission): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setCameraPermission):
(WTR::TestRunner::setMicrophonePermission):
(WTR::TestRunner::setUserMediaPermission):
(WTR::TestRunner::resetUserMediaPermission):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::setCameraPermission):
(WTR::TestController::setMicrophonePermission):
(WTR::TestController::decidePolicyForUserMediaPermissionRequestIfPossible):
(WTR::TestController::setUserMediaPermission): Deleted.
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/264149@main
